code/game/gamemodes/miniantags/bot_swarm/swarmer.dm

```diff
@@ -1,34 +1,67 @@
+/obj/item/device/unactivated_swarmer
 	name = "unactivated swarmer"
 	desc = "A currently unactivated swarmer. Swarmers can self activate at any time, it would be wise to immediately dispose of this."
 	icon = 'icons/mob/swarmer.dmi'
 	icon_state = "swarmer_unactivated"
 	origin_tech = "bluespace=4;materials=4;programming=7"
+	materials = list(MAT_METAL=100, MAT_GLASS=400)
 
-/obj/item/unactivated_swarmer/New()
-	notify_ghosts("An unactivated swarmer has been created in [get_area(src)]!", enter_link = "<a href=?src=[UID()];ghostjoin=1>(Click to enter)</a>", source = src, action = NOTIFY_ATTACK)
+/obj/item/device/unactivated_swarmer/New()
+	if(!crit_fail)
+		notify_ghosts("An unactivated swarmer has been created in [get_area(src)]!", enter_link = "<a href=?src=[UID()];ghostjoin=1>(Click to enter)</a>", source = src, action = NOTIFY_ATTACK)
 	..()
 
-/obj/item/unactivated_swarmer/Topic(href, href_list)
+/obj/item/device/unactivated_swarmer/Topic(href, href_list)
 	if(..())
 		return 1
 	if(href_list["ghostjoin"])
 		var/mob/dead/observer/ghost = usr
 		if(istype(ghost))
 			attack_ghost(ghost)
 
-/obj/item/unactivated_swarmer/attack_ghost(mob/user as mob)
+/obj/item/device/unactivated_swarmer/attackby(obj/item/weapon/W, mob/user, params)
+	..()
+	if(istype(W, /obj/item/weapon/screwdriver) && !crit_fail)
+		user.visible_message("<span class='warning'>[usr.name] deactivates [src].</span>",
+			"<span class='notice'>After some fiddling, you find a way to disable [src]'s power source.</span>",
+			"<span class='italics'>You hear clicking.</span>")
+		name = "deactivated swarmer"
+		desc = "A shell of swarmer that was completely powered down. It no longer can activate itself."
+		crit_fail = 1
+
+/obj/item/device/unactivated_swarmer/attack_ghost(mob/user as mob)
+
+	if(crit_fail)
+		to_chat(user, "<span class='warning'>This swarmer shell is completely depowered. You cannot activate it.</span>")
+		return
+
 	var/be_swarmer = alert("Become a swarmer? (Warning, You can no longer be cloned!)",,"Yes","No")
 	if(be_swarmer == "No")
 		return
+
+	if(crit_fail)//in case it depowers while ghost is looking at yes/no
+		to_chat(user, "<span class='warning'>This swarmer shell is completely depowered. You cannot activate it.</span>")
+		return
 	if(qdeleted(src))
 		to_chat(user, "Swarmer has been occupied by someone else.")
 		return
 	var/mob/living/simple_animal/hostile/swarmer/S = new /mob/living/simple_animal/hostile/swarmer(get_turf(loc))
 	S.key = user.key
 	qdel(src)
 
+/obj/item/device/unactivated_swarmer/swarmer_act(mob/living/simple_animal/hostile/swarmer/S)
+	if(S.resources + 50 > S.max_resources)
+		to_chat(S, "<span class='warning'>We have too many resources to reconsume this shell. Aborting.</span>")
+	else
+		..()
+		S.resources += 49 //refund the whole thing
+
+/obj/item/device/unactivated_swarmer/deactivated
+	name = "deactivated swarmer"
+	desc = "A shell of swarmer that was completely powered down. It no longer can activate itself."
```

Comment:
Grammatical tweak:

`It no longer can activate itself.` -> `It can no longer activate itself.`